src/modeler/AutoRest.Swagger/Validation/Extensions/LongRunningResponseValidationDelete.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Validation;
+using AutoRest.Core.Properties;
+using System;
+using System.Collections.Generic;
+using AutoRest.Swagger.Model;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class LongRunningResponseValidationDelete : LongRunningExtensionRule
+    {
+        /// <summary>
+        /// The template message for this Rule.
+        /// </summary>
+        public override string MessageTemplate => Resources.LongRunningResponsePostNotValid;
+
+        /// <summary>
+        /// An x-ms-long-running-operation extension passes this rule if the operation that this extension has a valid response defined.
+        /// </summary>
+        /// <param name="longRunning">long running extension object.</param>
+        /// <param name="context">Rule context.</param>
+        /// <param name="formatParameters">list of parameters for message formatter.</param>
+        /// <returns><c>true</c> if operation has valid response code modeled, otherwise <c>false</c>.</returns>
+        /// <remarks>This rule corresponds to M2005.</remarks>
+        public override bool IsValid(object longRunning, RuleContext context, out object[] formatParameters)
```

Comment:
If I understand correctly you're applying this rule at the ```Operation``` level, why not apply it at the ```Paths``` level and use Linq to identify operations that violate this rule, that would be much cleaner IMO